modules/tensor_mechanics/src/nodalkernels/NodalRotationalInertia.C

```diff
@@ -0,0 +1,225 @@
+
+    Real sum = x_orientation(0) * y_orientation(0) + x_orientation(1) * y_orientation(1) +
+               x_orientation(2) * y_orientation(2);
+
+    if (abs(sum) > 1e-4)
+      mooseError("NodalRotationalInertia: x_orientation and y_orientation should be perpendicular "
+                 "to each other.");
+
+    // Calculate z orientation as a cross product of the x and y orientations
+    RealGradient z_orientation;
+    z_orientation(0) = (x_orientation(1) * y_orientation(2) - x_orientation(2) * y_orientation(1));
+    z_orientation(1) = (x_orientation(2) * y_orientation(0) - x_orientation(0) * y_orientation(2));
+    z_orientation(2) = (x_orientation(0) * y_orientation(1) - x_orientation(1) * y_orientation(0));
+
+    // Rotation matrix from global to original beam local configuration
+    RankTwoTensor orientation;
+    orientation(0, 0) = x_orientation(0);
+    orientation(0, 1) = x_orientation(1);
+    orientation(0, 2) = x_orientation(2);
+    orientation(1, 0) = y_orientation(0);
+    orientation(1, 1) = y_orientation(1);
+    orientation(1, 2) = y_orientation(2);
+    orientation(2, 0) = z_orientation(0);
+    orientation(2, 1) = z_orientation(1);
+    orientation(2, 2) = z_orientation(2);
+
+    RankTwoTensor global_inertia = orientation.transpose() * _inertia * orientation;
+    _inertia = global_inertia;
+  }
+  else if ((isParamValid("x_orientation") && !isParamValid("y_orientation")) ||
+           (!isParamValid("x_orientation") && isParamValid("y_orientation")))
+    mooseError("NodalRotationalInertia: Both x_orientation and y_orientation should be provided if "
+               "x_orientation or "
+               "y_orientation is different from global x or y direction, respectively.");
+}
+
+Real
+NodalRotationalInertia::computeQpResidual()
+{
+  if (_dt == 0)
+    return 0;
+  else
+  {
+    const NumericVector<Number> & aux_sol_old = _aux_sys.solutionOld();
+
+    for (unsigned int i = 0; i < _nrot; ++i)
+    {
+      _rot_vel_old[i] =
+          aux_sol_old(_current_node->dof_number(_aux_sys.number(), _rot_vel_num[i], 0));
+      const Real rot_accel_old =
+          aux_sol_old(_current_node->dof_number(_aux_sys.number(), _rot_accel_num[i], 0));
+
+      _rot_accel[i] = 1.0 / _beta *
+                      ((((*_rot[i])[_qp] - (*_rot_old[i])[_qp]) / (_dt * _dt)) -
+                       _rot_vel_old[i] / _dt - rot_accel_old * (0.5 - _beta));
+      _rot_vel[i] =
+          _rot_vel_old[i] + (_dt * (1.0 - _gamma)) * rot_accel_old + _gamma * _dt * _rot_accel[i];
+    }
+
+    Real res = 0.0;
+    for (unsigned int i = 0; i < _nrot; ++i)
+      res += _inertia(_component, i) * (_rot_accel[i] + _rot_vel[i] * _eta * (1.0 + _alpha) -
+                                        _alpha * _eta * _rot_vel_old[i]);
+
+    return res;
+  }
+}
+
+Real
+NodalRotationalInertia::computeQpJacobian()
+{
+  if (_dt == 0)
+    return 0.0;
+  else
+    return _inertia(_component, _component) / (_beta * _dt * _dt) +
```

Comment:
Put an assert that _beta is greater than zero. It should be enforced by a range-checked param, but just to make sure that is enforced, use an assert as well.